mlem/core/data_type.py

```diff
@@ -230,6 +230,9 @@ def read(self, artifacts: Artifacts) -> DataType:
                 data = None
             elif res == "False":
                 data = False
+            elif self.data_type.ptype == "int":
+                # this handles cases such as int('0.0')
+                data = self.data_type.to_type(float(res))
```

Comment:
add tests for this case?

Reply:
I have removed this, this shouldn't be an issue
The fact that we were dealing with labels being automatically converted into arrays even when passed as None initially was the need for this condition, but I have removed it now